File: php/delete_contact.php
#!/usr/bin/env php
<?php

require_once 'api.php';

$api_key = $argv[1];
$access_token = $argv[2];
$contactid = $argv[3];

if (!$api_key || !$access_token) {
    echo "Usage: delete_contact.php <api_key> <access_token> <contactid>\n";
    exit();
}

$api = new VaultREAPI($api_key, $access_token);

// Delete contact
list($code, $result) = $api->delete('/contacts/' . $contactid);

if ($result) {
    echo $result;
}

if ($code == 403) {
    echo "HTTP 403: Forbidden";
    exit();
} elseif ($code == 401) {
    echo "HTTP 401: Invalid bearer token\n";
    exit();
} elseif ($code == 204) {
    echo "Record deleted successfully\n";
}


?>

File: perl/delete_contact.pl
#!/usr/bin/perl

use strict;
use warnings;

use FindBin;
use File::Spec;
use lib File::Spec->catdir($FindBin::Bin, '.');

use VaultREAPI;

my $api_key = $ARGV[0];
my $bearer_token = $ARGV[1];

if (!$api_key || !$bearer_token) {
    die "Usage: $0 <api_key> <bearer_token>\n";
}

my $api = new VaultREAPI($api_key, $bearer_token);

# Get user
my ($code, $result) = $api->delete('/contacts/1234');

print "HTTP Response Code: " . $code . "\n";

if ($result) {
    print $result;
}

exit;
